src/optimizer/transforms/char-case-insensitive-lowercase-transform.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * The MIT License (MIT)
+ * Copyright (c) 2017-present Dmitry Soshnikov <dmitry.soshnikov@gmail.com>
+ */
+
+'use strict';
+
+/**
+ * Transforms case-insensitive regexp to lowercase
+ *
+ * /AaBbÏ/i -> /aabbï/i
+ */
+module.exports = {
+  _AZClassRanges: null,
+  _hasUFlag: false,
+  init(ast) {
+    this._AZClassRanges = new Set();
+    this._hasUFlag = ast.flags.includes('u');
+  },
+  shouldRun(ast) {
+    return ast.flags.includes('i');
+  },
+  Char(path) {
+    const {node, parent} = path;
+    if (isNaN(node.codePoint)) {
+      return;
+    }
+
+    // Engine support for case-insensitive matching without the u flag
+    // for characters above \u1000 does not seem reliable.
+    if (!this._hasUFlag && node.codePoint >= 0x1000) {
+      return;
+    }
+
+    if (parent.type === 'ClassRange') {
+      // The only class ranges we handle are inside A-Z
+      if (isAZClassRange(parent)) {
+        this._AZClassRanges.add(parent);
+      }
+      if (!this._AZClassRanges.has(parent)) {
+        return;
+      }
+    }
+
+    const lower = node.symbol.toLowerCase();
+    if (lower !== node.symbol) {
+      node.value = displaySymbolAsValue(lower, node);
+      node.symbol = lower;
+      node.codePoint = lower.codePointAt(0);
+    }
+  }
+};
+
+function isAZClassRange(classRange) {
+  const {from, to} = classRange;
+  // A-Z
+  return from.codePoint >= 0x41 && from.codePoint <= 0x5a &&
```

Comment:
nit: I always keep forgetting the code points for these, maybe move them to constants above at module level.

```js
const UPPER_A_CP = 'A'.codePointAt(0);

...
// later in function:

return from.codePoint >= UPPER_A_CP ...
```

Reply:
Good idea, updated!